src/tools/ScenarioMeasurement/Startup/Startup.cs

```diff
@@ -114,28 +119,102 @@ static void checkArg(string arg, string name)
+
+            //Create wait funcs for steady state and post-compilation
+            Func<Process, bool> waitForSteadyState = metricType == MetricType.DotnetWatch ? (Proc) =>
+            {
+                StringBuilder output = new StringBuilder();
+                DataReceivedEventHandler stdOutProcessor = (s, e) =>
+                {
+                    if (!String.IsNullOrEmpty(e.Data))
+                    {
+                        output.AppendLine(e.Data);
+                        Console.WriteLine(e.Data);
+                    }
+                };
+                Proc.OutputDataReceived += stdOutProcessor;
+                Proc.BeginOutputReadLine();
+                bool isSteadyState = false;
+                int timeoutCount = 0;
+                while (!isSteadyState && timeoutCount < timeout)
+                {
+                    foreach (var line in output.ToString().Split(Environment.NewLine))
+                    {
+                        if (line.Contains("Application started."))
+                        {
+                            isSteadyState = true;
+                            break;
+                        }
+                    }
+                    timeoutCount++;
+                    Thread.Sleep(1000);
+                }
+                Proc.CancelOutputRead();
+                Proc.OutputDataReceived -= stdOutProcessor;
+                return true && timeoutCount < timeout;
+            }
+            : null;
+
+            Func<Process, ProcessHelper, bool> waitForRecompile = metricType == MetricType.DotnetWatch ? (Proc, innerLoopProcHelper) =>
+            {
+                StringBuilder output = new StringBuilder();
+                DataReceivedEventHandler stdOutProcessor = (s, e) =>
+                {
+                    if (!String.IsNullOrEmpty(e.Data))
+                    {
+                        output.AppendLine(e.Data);
+                        Console.WriteLine(e.Data);
+                    }
+                };
+                Proc.OutputDataReceived += stdOutProcessor;
+                Proc.BeginOutputReadLine();
+                bool isSteadyState = false;
+                int timeoutCount = 0;
+
+                while (!isSteadyState && timeoutCount < timeout)
+                {
+                    foreach (var line in output.ToString().Split(Environment.NewLine))
+                    {
+                        if (line.Contains("Hot reload of changes succeeded"))
+                        {
+                            isSteadyState = true;
+                            break;
+                        }
+                    }
+                    timeoutCount++;
+                    Thread.Sleep(1000);
+                }
+                Proc.CancelOutputRead();
+                Proc.OutputDataReceived -= stdOutProcessor;
+                return true && timeoutCount < timeout;
+            }
+            : null;
```

Comment:
These look to be basically the same method, can we refactor them so they take the string as input also?

Reply:
I was already thinking about doing this after I realized how close they are. Given that other tests would almost certainly want to do this same thing, we can start by this taking a string. If we want to eventually do more complex checking we can look into doing this as some sort of verification function that we pass in.